### DISCUSSION TITLE
postprocessor at arbitrary point values

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#26931] [Q&A General] [answered]
Dear all,

I have a multiapp where the master is a classical Moose FEM problem and the sub app is a set of scalar ODEs that are computed based on some transferred values from the master (from postprocessor values computed at given points). All works fine, though as it is coded now, the points must match the locations of the master mesh, while I would like to enable the computing of the postprocessor to be more flexible, let's say to compute those values at arbitrary points, that is, points that do not match a node in the mesh, though are inside the boundaries of the master domain. 

This said, I am wondering what would be the easiest and cleanest was to custom a pointvalue postprocessor that provides the value of an auxvariable (CONSTANT MONOMIAL) at an arbitrary point, possibly by interpolation from closest neighbors or from the value of the element that contains the point. 

Looking at the discussion, I realize that my post might be similar to ([#26747](https://github.com/idaholab/moose/discussions/26747)), though I am struggling to educate myself with functors. Could anyone provide any hint/suggestion/example which I could benefit from? That would be great.

Thanks,
mauro

**Comment by GiudGiud** [comment, marked as answer]
Hello

If there are may points, and you are looking to distribute in an arbitrary way, possibly programmatically, possibly from a file, I would suggest you use the Positions system.

This will let you keep track of the list of points and
- create multiapps at those locations
- sample the values of any field (all fields are functors) with this object
https://mooseframework.inl.gov/source/vectorpostprocessors/PositionsFunctorValueSampler.html

Guillaume

**Reply by mcacace** [reply]
Will have a look at it. Thanks

**Reply by mcacace** [reply]
It does seem to work quite nicely. Thanks @GiudGiud
